ironfish/src/rpc/routes/wallet/multisig/createIdentity.ts

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ParticipantSecret } from '@ironfish/rust-nodejs'
+import * as yup from 'yup'
+import { RPC_ERROR_CODES, RpcValidationError } from '../../../adapters/errors'
+import { ApiNamespace } from '../../namespaces'
+import { routes } from '../../router'
+import { AssertHasRpcContext } from '../../rpcContext'
+
+export type CreateIdentityRequest = {
+  name: string
+}
+
+export type CreateIdentityResponse = {
+  identity: string
+}
+export const CreateIdentityRequestSchema: yup.ObjectSchema<CreateIdentityRequest> = yup
+  .object({
+    name: yup.string().defined(),
+  })
+  .defined()
+
+export const CreateIdentityResponseSchema: yup.ObjectSchema<CreateIdentityResponse> = yup
+  .object({
+    identity: yup.string().defined(),
+  })
+  .defined()
+
+routes.register<typeof CreateIdentityRequestSchema, CreateIdentityResponse>(
+  `${ApiNamespace.wallet}/multisig/createIdentity`,
+  CreateIdentityRequestSchema,
+  async (request, context): Promise<void> => {
+    AssertHasRpcContext(request, context, 'wallet')
+
+    const { name } = request.data
+
+    if (await context.wallet.walletDb.hasMultisigSecret(name)) {
```

Comment:
Is this thread-safe (serializable consistency)? If I make two RPC calls to `createIdentity` with the same name at the same time, is it possible that both calls to `hasMultisigSecret` return false, hence resulting in the creation of 2 secrets, one of which will override the other?

Reply:
these calls should be wrapped in a database transaction